codersdk/users.go

```diff
@@ -104,15 +110,26 @@ type LoginWithPasswordResponse struct {
 	SessionToken string `json:"session_token" validate:"required"`
 }
 
+type OauthConversionResponse struct {
+	StateString string    `json:"state_string"`
+	ExpiresAt   time.Time `json:"expires_at" format:"date-time"`
+	ToLoginType LoginType `json:"to_login_type"`
+	UserID      uuid.UUID `json:"user_id" format:"uuid"`
+}
+
 type CreateOrganizationRequest struct {
 	Name string `json:"name" validate:"required,username"`
 }
 
 // AuthMethods contains authentication method information like whether they are enabled or not or custom text, etc.
 type AuthMethods struct {
-	Password AuthMethod     `json:"password"`
-	Github   AuthMethod     `json:"github"`
-	OIDC     OIDCAuthMethod `json:"oidc"`
+	// UserAuthenticationType returns the authentication method for the given
+	// caller if the request is an authenticated request. Otherwise it is empty.
+	UserAuthenticationType LoginType      `json:"me_login_type,omitempty"`
+	ConvertToOIDCEnabled   bool           `json:"convert_to_oidc_enabled"`
```

Comment:
Does this need to be in this payload as well? Seems like it only pertains to currently authed users

Reply:
This route is used to render the auth form on the FE. We now have an extra auth form for converting. Originally it was being pulled from the deployment options endpoint, but @BrunoQuaresma and I figured it'd be easier to just include all the info in 1 place.